vms/avm/service_test.go

```diff
@@ -6,17 +6,20 @@ package avm
 import (
 	"testing"
 
+	"github.com/ava-labs/gecko/snow/choices"
+
 	"github.com/ava-labs/gecko/database/memdb"
 	"github.com/ava-labs/gecko/ids"
 	"github.com/ava-labs/gecko/snow/engine/common"
+	"github.com/ava-labs/gecko/utils/formatting"
 	"github.com/ava-labs/gecko/vms/secp256k1fx"
 )
 
-func TestGetAssetDescription(t *testing.T) {
+func setup(t *testing.T) ([]byte, *VM, *Service) {
 	genesisBytes := BuildGenesisTest(t)
 
+	// TODO: Could this initialization be replaced by a call to GenesisVM()
```

Comment:
Yeah, the lock doesn't need to be held consistently across the usage of the VM. As long as the lock is held when an AVM function is called, then the call is valid. So this function could just wrap `GenesisVM` and return the service as well.

Reply:
Unfortunately this function couldn't wrap `GenesisVM`, because `GenesisVM` sets `vm.batchTimeout = 0` which caused the tests in this module to deadlock. There were other places that I managed to standardize though.

Reply:
That shouldn't happen. How does batchTimeout being set to 0 cause a deadlock?